src/models/tag.js

```diff
@@ -53,5 +53,31 @@ export default {
       (item) => { items.push(item.id) })
 
     return items
+  },
+
+  async merge(db, ids) {
+    let [keepId, ...mergeIds] = ids
+    let itemsWithOldTags = new Set()
+    // find all item ids which have taggings of tags to be deleted
+    for (const id of mergeIds) {
+      itemsWithOldTags.add(...await this.items(db, id))
```

Comment:
Using `this` here and below is potentially dangerous. It will commonly happen that these helpers are invoked directly and not via the common object used as the default export. We may also change the way we export these model functions.

Reply:
is there another way to access that function without using `this`? I don't understand the JS export system well enough to properly understand this issue

Reply:
The issue itself is not so much about the module/export system (but it's likely to be exposed because of it).

The problem is that `this` is only bound when the function is invoked (unless it's a bound or an arrow function!). If we reference `this.items` here, this may or may not work depending on how the function is called. The way use these functions makes it very likely that we'll have to pass them around to be invoked later, which would likely break. For example:

```js
let a = { f() { return this.g() }, g() { return '!' } }
a.f() //-> '!'
setTimeout(a.f, 25) //-> This will fail. It's similar to:
let f = a.f
f() //-> Which also fails.
```
It's very common that we have to pass these type of functions around (e.g., when we use them in Commands). The module/export dynamics also play a role, because named exports are the norm with ESM we may want to switch to named exports going forward so that we can `import { merge } from './models/tags.js'`.

In summary, you can easily address this by hoisting the `items` function similar to the `load` function (which is referenced by `create`). If we want to export all functions individually that's probably what we should do anyway, for all the model functions, but then we probably need address a few naming issues (e.g., the 'delete' function would clash with the keyword; 'items' in isolation is probably too cryptic etc..). Another alternative would be to still keep everything in a namespace object like we currently do, but also store it in a local variable for referencing purposes.

Reply:
In reference to the earlier example, of course `setTimeout(a.f.bind(a), 25)` would work, but it's not good if we always need to remember when to bind functions and when it's safe to pass them around.